--- 
title: "hydroITV"
subtitle: "Hydraulic trait inter- versus intra-specific local variability."
author: "Sylvain Schmitt"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
csl: biblio/mee.csl
bibliography: biblio/library.bib
link-citations: yes
colorlinks: yes
description: "Hydraulic trait inter- versus intra-specific local variability."
---
--- 
title: "hydroITV"
subtitle: "Hydraulic trait inter- versus intra-specific local variability."
author: "Sylvain Schmitt"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
csl: biblio/mee.csl
bibliography: biblio/library.bib
link-citations: yes
colorlinks: yes
description: "Hydraulic trait inter- versus intra-specific local variability."
---

`r if (knitr:::is_html_output()) '# Preface {-}'`

`r if (knitr:::is_html_output()) '![](https://img.shields.io/badge/stability-work_in_progress-lightgrey.svg)'`

`r if (knitr:::is_html_output()) 'This study aims to evaluate inter- and intra-specific variation in hydraulic and leaf traits in collaboration with Marion Boisseaux. We first plan to assess measurement and leaf variation to better study inter- and intra-specific variation. We plan to use a balanced and robust sampling of 10 species x 10 individuals per species representing 100 individuals. We first want to explore trait variation in covariation within and between species. If the results are convincing, we may want to further investigate variation within and between species, with topography controlling diameter variation. Overall, our study can provide a robust first assessment of variation in hydraulic traits within and between species for tropical tree species, with possible underlying drivers of this variation.'`

`r if (knitr:::is_html_output()) 'Have a nice reading.'`

`r if (knitr:::is_html_output()) '<div align="right">  *Sylvain*'`

<!--chapter:end:index.Rmd-->

# Introduction {-}

This study aims to evaluate inter- and intra-specific variation in hydraulic and leaf traits in collaboration with Marion Boisseaux. We first plan to assess measurement and leaf variation to better study inter- and intra-specific variation. We plan to use a balanced and robust sampling of 10 species x 10 individuals per species representing 100 individuals. We first want to explore trait variation in covariation within and between species. If the results are convincing, we may want to further investigate variation within and between species, with topography controlling diameter variation. Overall, our study can provide a robust first assessment of variation in hydraulic traits within and between species for tropical tree species, with possible underlying drivers of this variation. 

## ToDo

- [ ] `stan` models for balance sampling
- [ ] add protocols
- [ ] choose traits
- [ ] define sampling
- [ ] define calendar
- [ ] define logistic
- [ ] supervise
- [ ] analyze
- [ ] write

## Calendar

* **16/08**: define equipment needs
* **?/08**: assess measurement and leaf variation 
* **13-19/09**: field FTH
* **20-24/09**: analyses FTH

<!--chapter:end:01-Introduction.Rmd-->


# A not on balance sampling 

Placeholder


## Species and individual sampling
## Leaf variations and measurement errors

<!--chapter:end:02-BalanceSampling.Rmd-->


# Traits

Placeholder


## Soft traits
## Hydraulic traits
## Photochemical trait
## Measured traits
## Protocols

<!--chapter:end:03-Traits.Rmd-->

```{r setupinds, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(V.PhyloMaker)
library(ggfortify)
library(ggtree)
library(sf)
library(leaflet)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
crs <- '+proj=longlat +datum=WGS84' # leaflet CRS
```

# Individuals

This chapter define sampled species and individuals. 
Specifically, this chapter details:

* Collected species
* Individuals from the METRADICA project
* Candidates with varying DBH and TWI
* Field maps

## Species

We selected 10 species (Tab. \@ref(tab:speciesTab)) common with the METRADICA project representative of the tree phylogeny in Paracou (Fig. \@ref(fig:speciesPhylo)).

```{r speciesTab}
species <- data.frame(taxon = c(
  "Casearia_javitensis", 
  "Chrysophyllum_prieurii", 
  "Conceveiba_guianensis",
  "Gustavia_hexapetala",
  "Jacaranda_copaia subsp. copaia",
  "Laetia_procera",
  "Protium_stevensonii",
  "Tachigali_melinonii",
  "Virola_michelii",
  "Virola_surinamensis"
)) %>% 
  separate(taxon, c("Genus", "Species"), sep = "_", remove = F)
kable(select(species, Genus, Species), caption = "Studied species.")
```

```{r speciesPhylo, fig.cap="Selected species phylogeny."}
paracou <- read_csv2("data/Paracou_P16_2020.csv") %>% 
  dplyr::select(Family, Genus, Species) %>% 
  unique() %>% 
  full_join(species) %>% 
  mutate(species = paste(Genus, Species), genus = Genus, family = Family) %>% 
  dplyr::select(species, genus, family, taxon) %>% 
  mutate(taxon = as.character(taxon))
# tree <- phylo.maker(sp.list = paracou, tree = GBOTB.extended, nodes = nodes.info.1, scenarios = "S3")
# save(tree, file = "save/phylogeny.Rdata")
load("save/phylogeny.Rdata")
fortify(tree$scenario.3) %>% 
  mutate(species = gsub("_", " ", label)) %>% 
  left_join(paracou) %>% 
  ggtree(aes(col = taxon), layout="circular") + 
  geom_tiplab2(aes(alpha = !is.na(taxon), size = !is.na(taxon))) +
  theme_tree(legend.position='right', legend.text = element_text(face = "italic")) +
  scale_alpha_manual("taxon", values = c(0.2, 1)) +
  scale_size_manual("taxon", values = c(1, 2))
```

## METRADICA

This subparagraph details collected or wrong individuals in METRADICA (Tab. \@ref(tab:indMD)) and Marion's candidates (Tab. \@ref(tab:indMD), & Fig. \@ref(fig:indMDP16)),
but candidates of the project will be focused on P16 independently from her candidates and adjusted in the field. 

```{r indMD}
collected <- readxl::read_excel("data/202012_SelectionIndividus_V5.4.xlsx", sheet = "IndAMesurer") %>% 
  filter(TaxonActu %in% species$taxon) %>% 
  mutate(Statut = recode(Statut,
                         "AMesurerMD" = "Candidate",
                         "AMesurerMDPrioDK" = "Candidate",
                         "MesuréDK" = "Measured",
                         "MesuréDrought" = "Measured",
                         "MesuréMD" = "Measured",
                         "MesuréMD (MauvaiseSp)" = "Wrong",
                         "MesuréMD+Drought" = "Measured",
                         "MortMD" = "Wrong",
                         "MortParacou20" = "Wrong",
                         "MortParacou20PrioDK" = "Wrong",
                         "PasFaisable" = "Wrong",
                         ))
collected %>% 
  group_by(TaxonActu, Statut) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(TaxonActu ~ Statut, value.var = "N") %>% 
  mutate(TaxonActu = gsub("_", " ", TaxonActu)) %>% 
  kable(caption = "Sumary of individuals (from 202012_SelectionIndividus_V5.4.xlsx)")
```

```{r indMB}
tocollect <- readxl::read_excel("data/FTH2021_SelectionIndividus_V5.5.xlsx", sheet = "IndAMesurer") %>% 
  filter(TaxonActu %in% species$taxon) %>% 
  st_as_sf(coords = c("Xutm", "Yutm"),
           crs = '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0') %>% 
  mutate(DBH = Circ/pi)
tocollect %>% 
  group_by(TaxonActu, Plot) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(TaxonActu ~ Plot, value.var = "N") %>% 
  mutate_all(funs(ifelse(is.na(.), " ", .))) %>% 
  kable(caption = "Sumary of candidates (from FTH2021_SelectionIndividus_V5.5.xlsx)")
```


```{r indMDP16, fig.cap="Marion's candidates in P16."}
palSp <- colorFactor("viridis", tocollect$TaxonActu)
limits <- st_read("data/OverallPlots/OverallPlots.shp", quiet = T)
leaflet(data = st_transform(tocollect, crs = crs)) %>%
  addTiles() %>% 
  addPolylines(data = st_transform(limits, crs = crs), col = "grey") %>% 
  addCircles(color = ~palSp(TaxonActu), opacity = 1,
             label = ~ TaxonActu, radius = ~ DBH/5) %>% 
  addLegend(pal = palSp, values = ~ TaxonActu)
```

## Candidates

I selected 15 individuals per species in P16 (10 + 5 extras) minimizing DBH variation while maximizing TWI variation (Fig. \@ref(fig:candFig), Tab. \@ref(tab:candTab), Fig. \@ref(fig:candMap)). We are lacking individuals for *Casearia javitensis*, *Protium stevensonii*, and *Virola surinamensis*.

```{r candidates}
candidates <- read_csv2("data/Paracou_P16_2020.csv") %>%
  left_join(species) %>%
  filter(!is.na(taxon)) %>%
  mutate(DBH = CircCorr/pi) %>% 
  st_as_sf(coords = c("Xutm", "Yutm"),
           crs = '+proj=utm +zone=22 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0')
candidates$TWI <- raster::extract(raster::raster("data/TWI_1m.tif"), candidates)
candidates <- candidates %>% 
  group_by(taxon) %>% 
  mutate(N = n()) %>% 
  mutate(cand0 = ifelse(N <= 17, 1, 0)) %>% 
  mutate(m = median(DBH), l = quantile(DBH, 0.3), u = quantile(DBH, 0.6)) %>% 
  mutate(cand0 = ifelse(cand0 == 0 & DBH >= l & DBH <= u, 1, cand0))
cand <- as.data.frame(candidates) %>% 
  group_by(taxon) %>% 
  filter(cand0 == 1, N > 15) %>% 
  sample_n(15) %>% 
  unique()
candidates <- candidates %>% 
  mutate(candidate =  ifelse(idTree %in% cand$idTree, 1, 0)) %>% 
  mutate(candidate =  ifelse(N <= 15, 1, candidate)) %>% 
  mutate(candidate =  as.factor(candidate)) %>% 
  select(-cand0)
```

```{r candFig, fig.cap="DBH and TWI of candidates in P16."}
ggplot(candidates, aes(DBH, TWI, col = candidate, alpha = candidate)) +
  geom_point() +
  facet_wrap(~ taxon, scales = "free_x") +
  scale_color_manual("Candidate", values = c("red", "grey")) +
  scale_alpha_manual("Candidate", values = c(1, 0.5)) +
  theme(legend.position = c(0.8, 0.1)) +
  xlab("Diameter at breast height (DBH, cm)") +
  ylab("Topographic wetness index (TWI)")
```

```{r candTab}
as.data.frame(candidates) %>% 
  filter(candidate == 1) %>% 
  group_by(taxon, N) %>% 
  summarise(Nsel = n()) %>% 
  select(taxon, Nsel, N) %>% 
  kable(caption = "Candidates in P16.")
```

```{r candMap, fig.cap="Candidates in P16."}
palSp <- colorFactor("viridis", candidates$taxon)
leaflet(data = st_transform(filter(candidates, candidate == 1), 
                            crs = crs)) %>%
  addTiles() %>% 
  addPolylines(data = st_transform(filter(limits, Plot == 16), 
                                   crs = crs), col = "grey") %>% 
  addCircles(color = ~palSp(taxon), opacity = 1,
             label = ~ taxon, radius = ~ DBH/5) %>% 
  addLegend(pal = palSp, values = ~ taxon)
```

## Maps

Maps are automatically built using `sf` and `ggplot2` and save in the folder `maps`:

* [P16 North-West](maps/P16NW.png)
* [P16 North-East](maps/P16NE.png)
* [P16 South-West](maps/P16SW.png)
* [P16 South-East](maps/P16SE.png)

```{r mapsFun}
contour <- st_read("data/ContourLinesPlots/ContourLinePlots.shp", quiet = T)
crs_rot = "+proj=omerc +lat_0=36.934 +lonc=-90.849 +alpha=0 +k_0=.7 +datum=WGS84 +units=m +no_defs +gamma=20"
make_map <- function(file, title, subplots){
  sublimits <- filter(limits, Plot == 16, Subplot %in% subplots) %>% 
    st_transform(crs = crs_rot)
  subcontour <- st_transform(contour, crs = crs_rot) %>% st_crop(sublimits)
  subcandidates <- filter(candidates, candidate == 1, SubPlot %in% subplots) %>% 
    st_transform(crs = crs_rot) %>% 
    mutate(label = paste0("P", Plot, "-", SubPlot, "-", TreeFieldNum, "_", substr(Genus, 1, 3), substr(Species, 1, 3), "_", round(DBH)))
  
  ggplot() +
    geom_sf(data = subcontour, fill = NA, col = "lightgrey") +
    geom_sf(data = sublimits, fill = NA, col = "darkgrey", aes(text = subplots)) +
    geom_sf_text(data = sublimits, aes(label = Subplot), colour = "darkgrey") +
    geom_sf(data = subcandidates, col = "black") +
    ggrepel::geom_text_repel(
      data = subcandidates,
      aes(label = label, geometry = geometry),
      stat = "sf_coordinates",
      min.segment.length = 0
    ) +
    theme(axis.title = element_blank(), axis.text = element_blank(), 
          axis.ticks = element_blank(), axis.line = element_blank()) +
    scale_color_discrete(guide = "none") +
    ggtitle(title) +
    ggsave(file = file, path = 'maps', width = 297, height = 420, unit = 'mm', dpi = 300)
}
```

```{r maps, eval=F}
make_map("P16NW.png", "Plot 16 North-West", c(1:3, 6:8, 11:13))
make_map("P16NE.png", "Plot 16 North-East", c(3:5, 8:10, 13:15))
make_map("P16SW.png", "P16 South-West", c(11:13, 16:18, 21:23))
make_map("P16SE.png", "P16 South-East", c(13:15, 18:20, 23:25))
```

```{r}
include_graphics("maps/P16NW.png")
include_graphics("maps/P16NE.png")
include_graphics("maps/P16SW.png")
include_graphics("maps/P16SE.png")
```


<!--chapter:end:04-Individuals.Rmd-->

# Traits covariation

Subsequent analysis aimed to explore co-variations of individual traits. 
Specifically, we investigated individual traits co-variation at several taxonomic scales: among species, and within species.

**Waiting data, inspiration can be found here: https://github.com/sylvainschmitt/PhD/blob/master/documents/functional/01-PCA.Rmd .**

<!--chapter:end:05-Covariations.Rmd-->

# Traits covariation

Subsequent analysis aimed to assess variance partitioning among and within species.

**Waiting data.**

<!--chapter:end:06-VariancePartitioning.Rmd-->

# Environment & Ontogeny

Subsequent analysis aimed to select functional traits descriptors in order to further model traits variation. 

Phenotypic variation, and therefore functional traits variation, will be shaped (i) by genetic heritage, through genotypes, (ii) by the environment (both abiotic and biotic) with spatial and temporal heterogeneity, and (iii) by random stochastic factors [@Whitlock2007]. Our analysis did not include yet raw genetic data. Consequently, genetic heritage will be both represented by taxonomic levels (species and genera) and ontogenetic factors. Environment will be split between abiotic environment and biotic interactions, here restrained to tree to tree interactions. We will not explore tree interactions with their environment trough herbivory and pathogens, besides their huge impact on tree life history [but see @VanderPutten2001].

We selected functional traits descriptors by studying descriptors co-variation and keeping only little correlated variables with the best ecological meaning to represent (i) ontogeny, (ii) abiotic environment, and (iii) biotic interactions.

**Waiting data, inspiration can be found here: https://github.com/sylvainschmitt/PhD/blob/master/documents/functional/02-Environment.Rmd .**

<!--chapter:end:07-Environments.Rmd-->

# Models

The subsequent analysis aimed to graphically explore variations in leaf traits according to 4 descriptors: (i) tree ontogeny, (ii) biotic interactions and (iii) abiotic environment, and (iv) phylogeny through taxonomic levels. From these observations we developed a generic model linking individual phenotype to its descriptors. The generic model will be inferred later using Bayesian inference.

**Waiting data, inspiration can be found here: https://github.com/sylvainschmitt/PhD/blob/master/documents/functional/03-Models.Rmd .**

<!--chapter:end:08-Models.Rmd-->


# Literature

Placeholder


## @Hajek2016
## @Lobo2018
## @Westerband2021
## @Martinez-Vilalta2009
## @Rosas2019

<!--chapter:end:90-literature.Rmd-->

# References {-}

<!--chapter:end:99-References.Rmd-->

